sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/Profiling/ProfilingSessionEventSource.cs

```diff
@@ -83,6 +125,88 @@ protected override void OnEventCommand(EventCommandEventArgs command)
         /// Gets or sets the active profiling session ID.
         /// The value will be null if there is no active session.
         /// </summary>
-        public string? SessionId { get; private set; }
+        public string? SessionId
+        {
+            [NonEvent]
+            get => _sessionId;
+
+            [NonEvent]
+            private set
+            {
+                if (_sessionId != value)
+                {
+                    _sessionId = value;
+                    _sessionIdChanged?.Invoke(this, value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Occurs when the session ID has changed, meaning either a session has started or stopped.
+        /// </summary>
+        /// <remarks>
+        /// If you subscribe to this event when a session has already started, then your handler
+        /// will be called immediately.
+        /// </remarks>
+        public event EventHandler<string?> SessionIdChanged
+        {
+            [NonEvent]
+            add
+            {
+                _sessionIdChanged += value;
+
+                // New subscribers are notified of an active session right away.
+                if (SessionId != null)
+                {
+                    value.Invoke(this, SessionId);
+                }
+            }
+
+            [NonEvent]
+            remove
+            {
+                _sessionIdChanged -= value;
+            }
+        }
+
+        /// <summary>
+        /// Write the attributes of a <see cref="Resource"/>.
+        /// </summary>
+        /// <param name="resource">The resource.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="resource"/> is null.</exception>
+        [NonEvent]
```

Comment:
Can remove this comment. No longer throwing exception.

```suggestion
        [NonEvent]
```